src/test/java/hudson/matrix/MatrixProjectTest.java

```diff
@@ -569,7 +569,18 @@ public void dontRunOnExclusiveSlave() throws Exception {
             projects.add(m);
         }
 
-        DumbSlave s = new DumbSlave("big", "this is a big slave", j.createTmpDir().getPath(), "20", EXCLUSIVE, "", j.createComputerLauncher(null), RetentionStrategy.NOOP);
+        TemporaryFolder tf = new TemporaryFolder();
```

Comment:
This won't work properly unless you make it a `@Rule` field.

Reply:
Good catch!